hystrix-contrib/hystrix-javanica/build.gradle

```diff
@@ -1,10 +1,104 @@
+apply plugin: 'java'
+
+configurations {
+    ajtools
+    ajcTestCompile.extendsFrom testCompile
+    ajcTestRuntime.extendsFrom testRuntime
+    ajc
+}
+
+
+sourceSets {
+    ajcTest {
+        compileClasspath += main.output + test.output
+        runtimeClasspath += main.output + test.output
+    }
+
+    ajc {
+        def main = sourceSets.main
+        java.srcDirs = main.java.srcDirs
+        resources.srcDirs = main.resources.srcDirs
+        compileClasspath = main.compileClasspath
+        runtimeClasspath = main.runtimeClasspath
+    }
+}
+
+
+
+compileAjcTestJava {
+
+    sourceCompatibility = "1.6"
+    targetCompatibility = "1.6"
+
+    doLast {
+        ant.taskdef(resource: "org/aspectj/tools/ant/taskdefs/aspectjTaskdefs.properties", classpath: configurations.ajtools.asPath)
+        ant.iajc(source: "1.6", target: "1.6",
+                destDir: "${sourceSets.ajcTest.output.classesDir.absolutePath}", maxmem: "512m", fork: "true",
+                classpath: "${sourceSets.main.output.classesDir.absolutePath};${configurations.testCompile.asPath}")
+                {
+                    sourceroots {
+                        files("src/ajcTest/java/com/netflix/hystrix/contrib/javanica/test/aspectj", "src/test/java/com/netflix/hystrix/contrib/javanica/test/common", "src/main/java/com/netflix/hystrix/contrib/javanica/aop/aspectj").each {
+                            File file -> pathelement(location: file.absolutePath)
+                        }
+                    }
+                }
+    }
+
+}
+
+compileAjcJava {
+    sourceCompatibility = "1.6"
+    targetCompatibility = "1.6"
+
+    doLast {
+        ant.taskdef(resource: "org/aspectj/tools/ant/taskdefs/aspectjTaskdefs.properties", classpath: configurations.ajtools.asPath)
+        ant.iajc(source: "${sourceCompatibility}", target: "${targetCompatibility}",
+                destDir: "${sourceSets.ajc.output.classesDir.absolutePath}", maxmem: "512m", fork: "true", "showWeaveInfo": "true",
+                classpath: "${configurations.compile.asPath}")
+
+                {
+                    sourceroots {
+                        sourceSets.ajc.java.srcDirs.each {
+                            pathelement(location: it.absolutePath)
+
+                        }
+                    }
+                }
+    }
+}
+
+task ajcTest(type: Test) {
```

Comment:
@mattrjacobs I added this task that creates jar with aspects compiled using ajc, so now build task results to two jars:
hystrix-javanica.jar
hystrix-javanica-ctw.jar

Should we change something else to publish both artifacts correctly ?